### PR TITLE
EVAKA-4213 Income statement (citizen)

### PR DESCRIPTION
**Pull request by akheron**

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+private fun <This : SqlStatement<This>> SqlStatement<This>.bindEntrepreneur(entrepreneur: Entrepreneur): This =
+    run { if (entrepreneur.selfEmployed != null) bindSelfEmployed(entrepreneur.selfEmployed) else this }
+        .run { if (entrepreneur.limitedCompany != null) bindLimitedCompany(entrepreneur.limitedCompany) else this }
+        .bind("fullTime", entrepreneur.fullTime)
+        .bind("startOfEntrepreneurship", entrepreneur.startOfEntrepreneurship)
+        .bind("spouseWorksInCompany", entrepreneur.spouseWorksInCompany)
+        .bind("startupGrant", entrepreneur.startupGrant)
+        .bind("partnership", entrepreneur.partnership)
+        .bind("lightEntrepreneur", entrepreneur.lightEntrepreneur)
+        .run { if (entrepreneur.accountant != null) bindAccountant(entrepreneur.accountant) else this }
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt:222]: I don't know if you looked into it, but `bindKotlin` could reduce a lot of this code, especially the places where an identically named field is bound.

**Review by jaahon — COMMENTED** [review]
This PR is kind of too long to review properly, but mostly everything looked very good 😄. 

It seems like scrolling to errors doesn't work when trying to save an incomplete form.

File: service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt (hunk 1 above)
Comment (akheron) [reply service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt:222]: Didn't get `bindKotlin` to work 🤷

File: frontend/src/citizen-frontend/income-statements/IncomeStatements.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      <FixedSpaceColumn>
+        <FixedSpaceRow>
+          <Label>Tulotiedot päivitetty</Label>
+          <div>{incomeStatement.startDate.format()}</div>
+        </FixedSpaceRow>
+        <Button
+          text="Muokkaa"
+          onClick={() => history.push(`/income/${incomeStatement.id}`)}
+        />
+      </FixedSpaceColumn>
```
[the remark this answers is not in the document]
Comment (akheron) [reply frontend/src/citizen-frontend/income-statements/IncomeStatements.tsx:94]: This list will be redesigned any case in a later PR, will fix then.

File: frontend/src/citizen-frontend/income-statements/IncomeStatementForm.tsx [hunk 3]
```diff
@@ -0,0 +1,1103 @@
...
+  formData: Form.Entrepreneur
+): Form.Entrepreneur {
+  // if (formData.incomeType.startsWith('ENTREPRENEUR_SELF_EMPLOYED_')) {
+  //   return { ...formData, incomeSource: null }
+  // }
+  // if (formData.incomeType === 'ENTREPRENEUR_PARTNERSHIP') {
+  //   return { ...formData, incomeSource: null }
+  // }
+  return formData
+}
```
[the remark this answers is not in the document]
Comment (akheron) [reply frontend/src/citizen-frontend/income-statements/IncomeStatementForm.tsx]: Oops, removed 👍

**Comment by akheron** [comment]
Fixed scrolling to errors on Firefox 👍

**Review by jaahon — APPROVED** [review]
🏆

**Pull request merged**